source/macos/AdaptiveCards/AdaptiveCardsTests/Renderers/ColumnRendererTests.swift

```diff
@@ -103,44 +103,77 @@ class ColumnRendererTests: XCTestCase {
         // No padding added
         column = FakeColumn.make(isVisible: true, style: .none)
-        let columnView = renderColumnView()
+        let columnView = renderColumnView(column)
         XCTAssertEqual(columnView.stackView.arrangedSubviews.count, 1)
     }
     
     func testColumnWithHeight() {
         column = FakeColumn.make(isVisible: true, style: .accent, height: .stretch)
-        let columnView = renderColumnView()
+        let columnView = renderColumnView(column)
         XCTAssertEqual(columnView.stackView.arrangedSubviews.count, 1)
     }
     
     func testEmptyColumnRendersWithoutError() {
         column = .make(items: [])
         
-        XCTAssertNoThrow(renderColumnView())
+        XCTAssertNoThrow(renderColumnView(column))
     }
     
-    private func renderColumnView() -> ACRColumnView {
-        let view = columnRenderer.render(element: column, with: hostConfig, style: .default, rootView: FakeRootView(), parentView: NSView(), inputs: [], config: .default)
+    func testRendererInheritsVerticalContentAlignment() {
+        var parentColumn = FakeColumn.make(verticalContentAlignment: .top)
+        var parentColumnView = renderColumnView(parentColumn)
+        column = .make(items: [FakeTextBlock.make()], verticalContentAlignment: .nil)
+        var columnView = renderColumnView(column, parentView: parentColumnView)
+        
+        // For HeightType Property we have add stretchable view. so it will increase count for subviews.
+        XCTAssertEqual(columnView.stackView.arrangedSubviews.capacity, 2)
+        
+        parentColumn = FakeColumn.make(verticalContentAlignment: .center)
+        parentColumnView = renderColumnView(parentColumn)
+        columnView = renderColumnView(column, parentView: parentColumnView)
+        
+        // SpaceView 2
+        XCTAssertEqual(columnView.stackView.arrangedSubviews.capacity, 4)
+        
+        parentColumn = FakeColumn.make(verticalContentAlignment: .bottom)
+        parentColumnView = renderColumnView(parentColumn)
+        columnView = renderColumnView(column, parentView: parentColumnView)
+        // since we removed padding view all together and replaced with lastPadding but use paddingView for vertical content alignment this has changed
+        // SpaceView 1
+        XCTAssertEqual(columnView.stackView.arrangedSubviews.capacity, 3)
+    }
+    
+    func testInvisibleViewsWithVerticalContentAlignment() {
+        let stretcableView = FakeColumn.make(height: .stretch)
+        column = .make(items: [stretcableView], verticalContentAlignment: .bottom)
+        let columnView = renderColumnView(column)
+        
+        // Normally there are 3 subviews in case of botom, with stretcable view, it is reduced to 2
+        XCTAssertEqual(columnView.stackView.arrangedSubviews.capacity, 2)
+    }
+    
+    private func renderColumnView(_ element: ACSColumn, parentView: NSView = NSView()) -> ACRColumnView {
```

Comment:
Why are we add element parameter? if we can take it from global `column`.

Reply:
In case we want to render a second column we will have to keep modifying the global variable and then render and then modify the global variable and render again, so we have it this way. I agree we can remove the global variable ideally then we just have to declare column in every test